src/interfacers/EmonHubKNXInterfacer.py

```diff
@@ -0,0 +1,273 @@
+                               'nodename':'KNX',
+                               'validate_checksum': True,
+                               'meters':[]}
+
+        self._last_read_time = 0
+
+        try:
+            self.loop = asyncio.get_event_loop()
+
+            task = self.loop.create_task(self.initKnx(gateway_ip, local_ip))
+            self.loop.run_until_complete(task)
+
+            self.cargoList = {}
+
+        except ModuleNotFoundError as err:
+            self._log.error(err)
+            self.ser = False
+
+
+    def action(self):
+        super().action()
+
+
+    async def initKnx(self, gateway_ip, local_ip):
+        connection_config = ConnectionConfig(
+                                        connection_type=ConnectionType.TUNNELING,
+                                        gateway_ip=gateway_ip,
+                                        local_ip = local_ip
+        )
+
+        self._log.debug("Connect to KNX Gateway : " + gateway_ip)
+        self.xknx = XKNX(connection_config=connection_config,   connection_state_changed_cb = self.connection_state_changed_cb,device_updated_cb=self.device_updated_cb, daemon_mode=False)
+
+    async def startKnx(self):
+        try:
+            await self.xknx.start()
+        except Exception as err: 
+            self._log.error("KNX Error start:")
+            self._log.error(err);
+
+    def connection_state_changed_cb(self, state):
+        self._log.debug("KNX CnxUpdate:" )
+        self._log.debug(state)
+        
+
+    def device_updated_cb(self, device):
+        value = device.resolve_state()
+        name = device.name
+        unit = device.unit_of_measurement()
+
+        self._log.info("Device:" + name + ' <> ' + str(value))
+
+        pos = name.index("_")
+        meter = name[0:pos]
+        key = name[pos+1:]
+
+        meterObj=self._settings['meters'][meter]
+        dptConf = meterObj[key]
+        if 'divider' in dptConf:
+            divider = dptConf["divider"]
+            if divider != '':
+                value = float(value) / float(divider)
+
+
+
+        result = {}
+        result[key] = [value,unit]
+
+
+        if meter in self.cargoList:
+            c = self.cargoList[meter]
+            self.add_result_to_cargo(c, result)
+        else:
+            cargoNew = Cargo.new_cargo("", False,[], [])
+            cargoNew.nodeid = meter
+            self.cargoList[meter] = cargoNew
+            self.add_result_to_cargo(cargoNew, result)
+
+
+
+    def add_result_to_cargo(self, cargo, result):
+        if result != None:
+
+            for key in result:
+                cargo.names.append(key)
+                cargo.realdata.append(result[key][0])
+        else:
+            self._log.info("Decoded KNX data: None")
+
+
+    async def setupSensor(self):
+        metters = self._settings['meters']
+
+        self.sensor={}
+        for metter in metters:
+            dpPoint = metters[metter]
+
+            for dpKey in dpPoint:
```

Comment:
same I think you can iterate with items()

use var name like `dp_config` (snake case) and not `dpConfig` (camel case). the OEM maintainers are not using pylint to analyse their code but I think they should :-) I think the xknx lib is following this pattern and I think emonhub began like that, see the emonhub_interfacer base code